Could not determine the canonical constructor when quarkus.rest.jackson.optimization.enable-reflection-free-serializers=true

### Describe the bug

Enabling `quarkus.rest.jackson.optimization.enable-reflection-free-serializers=true` causes build failures when using Java records with annotated components. The Jandex indexer fails to match canonical constructor parameter types with record components due to annotation presence.

### Expected behavior

Successful build with reflection-free serialization supporting records containing annotated components.


### Actual behavior

Build failure due to type mismatch in Jandex index between annotated parameters and record components.


### How to Reproduce?

1. Create record with annotated component:
```
import jakarta.validation.constraints.Min;

public record CreateRequest(String is, @Min(16) int length)
```
2. Build with Maven/Gradle

**Observed Error:**
```
[ERROR] Failed to execute goal io.quarkus:quarkus-maven-plugin:3.18.2:build (default) on project simple-rest: Failed to build quarkus application: io.quarkus.builder.BuildException: Build failure: Build failed due to errors
[ERROR] 	[error]: Build step io.quarkus.resteasy.reactive.jackson.deployment.processor.ResteasyReactiveJacksonProcessor#handleEndpointParams threw an exception: java.lang.IllegalStateException: Could not determine the canonical constructor
[ERROR] 	at org.jboss.jandex.ClassInfo.canonicalRecordConstructor(ClassInfo.java:799)
[ERROR] 	at io.quarkus.resteasy.reactive.jackson.deployment.processor.JacksonDeserializerFactory.createDeserializedObject(JacksonDeserializerFactory.java:238)
[ERROR] 	at io.quarkus.resteasy.reactive.jackson.deployment.processor.JacksonDeserializerFactory.createSerializationMethod(JacksonDeserializerFactory.java:213)
[ERROR] 	at io.quarkus.resteasy.reactive.jackson.deployment.processor.JacksonCodeGenerator.create(JacksonCodeGenerator.java:86)
[ERROR] 	at io.quarkus.resteasy.reactive.jackson.deployment.processor.JacksonCodeGenerator.create(JacksonCodeGenerator.java:67)
[ERROR] 	at io.quarkus.resteasy.reactive.jackson.deployment.processor.ResteasyReactiveJacksonProcessor.handleEndpointParams(ResteasyReactiveJacksonProcessor.java:430)
[ERROR] 	at java.base/java.lang.invoke.MethodHandle.invokeWithArguments(MethodHandle.java:733)
[ERROR] 	at io.quarkus.deployment.ExtensionLoader$3.execute(ExtensionLoader.java:856)
[ERROR] 	at io.quarkus.builder.BuildContext.run(BuildContext.java:256)
[ERROR] 	at org.jboss.threads.ContextHandler$1.runWith(ContextHandler.java:18)
[ERROR] 	at org.jboss.threads.EnhancedQueueExecutor$Task.doRunWith(EnhancedQueueExecutor.java:2675)
[ERROR] 	at org.jboss.threads.EnhancedQueueExecutor$Task.run(EnhancedQueueExecutor.java:2654)
[ERROR] 	at org.jboss.threads.EnhancedQueueExecutor.runThreadBody(EnhancedQueueExecutor.java:1627)
[ERROR] 	at org.jboss.threads.EnhancedQueueExecutor$ThreadBody.run(EnhancedQueueExecutor.java:1594)
[ERROR] 	at java.base/java.lang.Thread.run(Thread.java:1583)
[ERROR] 	at org.jboss.threads.JBossThread.run(JBossThread.java:499)
[ERROR] -> [Help 1]
```

### Output of `uname -a` or `ver`

Darwin Kernel Version 24.3.0: root:xnu-11215.81.4~3/RELEASE_ARM64_T6020 arm64

### Output of `java -version`

 Corretto-21.0.6.7.1

### Quarkus version or git rev

3.18.2

### Build tool (ie. output of `mvnw --version` or `gradlew --version`)

mvn

### Additional information

**Code Analysis:**
The failure occurs in Jandex's `org.jboss.jandex.ClassInfo#canonicalRecordConstructor()` when comparing:
`if (!parameters[i].equals(recordComponents[recordComponentPositions[i] & 0xFF].type())) {`
- Parameter type: `int` (unannotated)
- Record component type: `@Min(16) int` (annotated)
So the `int` is not equal `@Min(16) int`

PS. I am not sure if it related to the quarkus project but in any case `enable-reflection-free-serializers` is not working

--- a/bom/application/pom.xml
+++ b/bom/application/pom.xml
@@ -22,7 +22,7 @@
         <cyclonedx.version>9.0.5</cyclonedx.version>
         <expressly.version>5.0.0</expressly.version>
         <findbugs.version>3.0.2</findbugs.version>
-        <jandex.version>3.2.4</jandex.version>
+        <jandex.version>3.2.5</jandex.version>
         <javax.annotation-api.version>1.3.2</javax.annotation-api.version>
         <javax.inject.version>1</javax.inject.version>
         <parsson.version>1.1.7</parsson.version>

--- a/build-parent/pom.xml
+++ b/build-parent/pom.xml
@@ -29,7 +29,7 @@
         <jboss-bridger-plugin.version>1.6.Final</jboss-bridger-plugin.version>
 
         <!-- Jandex versions -->
-        <jandex.version>3.2.4</jandex.version>
+        <jandex.version>3.2.5</jandex.version>
         <jandex-gradle-plugin.version>1.0.0</jandex-gradle-plugin.version>
 
         <asciidoctorj.version>2.5.13</asciidoctorj.version>

--- a/independent-projects/arc/pom.xml
+++ b/independent-projects/arc/pom.xml
@@ -45,7 +45,7 @@
         <version.jta>2.0.1</version.jta>
         <!-- main versions -->
         <version.gizmo>1.8.0</version.gizmo>
-        <version.jandex>3.2.4</version.jandex>
+        <version.jandex>3.2.5</version.jandex>
         <version.jboss-logging>3.6.1.Final</version.jboss-logging>
         <version.mutiny>2.8.0</version.mutiny>
         <version.bridger>1.6.Final</version.bridger>

--- a/independent-projects/bootstrap/pom.xml
+++ b/independent-projects/bootstrap/pom.xml
@@ -34,7 +34,7 @@
         <javax.annotation-api.version>1.3.2</javax.annotation-api.version>
         <javax.inject.version>1</javax.inject.version>
         <project.build.sourceEncoding>UTF-8</project.build.sourceEncoding>
-        <jandex.version>3.2.4</jandex.version>
+        <jandex.version>3.2.5</jandex.version>
         <jmh.version>1.37</jmh.version>
 
         <!-- Dependency versions -->

--- a/independent-projects/junit5-virtual-threads/pom.xml
+++ b/independent-projects/junit5-virtual-threads/pom.xml
@@ -41,7 +41,7 @@
         <compiler.plugin.version>3.13.0</compiler.plugin.version>
         <enforcer.plugin.version>3.2.1</enforcer.plugin.version>
         <surefire.plugin.version>3.5.2</surefire.plugin.version>
-        <jandex.version>3.2.4</jandex.version>
+        <jandex.version>3.2.5</jandex.version>
 
         <junit.jupiter.version>5.10.5</junit.jupiter.version>
         <junit.testkit.version>1.10.3</junit.testkit.version>

--- a/independent-projects/qute/pom.xml
+++ b/independent-projects/qute/pom.xml
@@ -40,7 +40,7 @@
         <project.build.sourceEncoding>UTF-8</project.build.sourceEncoding>
         <version.junit>5.10.5</version.junit>
         <version.assertj>3.27.3</version.assertj>
-        <version.jandex>3.2.4</version.jandex>
+        <version.jandex>3.2.5</version.jandex>
         <version.gizmo>1.8.0</version.gizmo>
         <version.jboss-logging>3.6.1.Final</version.jboss-logging>
         <version.smallrye-mutiny>2.8.0</version.smallrye-mutiny>

--- a/independent-projects/resteasy-reactive/pom.xml
+++ b/independent-projects/resteasy-reactive/pom.xml
@@ -45,7 +45,7 @@
         <project.build.sourceEncoding>UTF-8</project.build.sourceEncoding>
         <!-- Versions -->
         <jakarta.enterprise.cdi-api.version>4.1.0</jakarta.enterprise.cdi-api.version>
-        <jandex.version>3.2.4</jandex.version>
+        <jandex.version>3.2.5</jandex.version>
         <bytebuddy.version>1.15.11</bytebuddy.version>
         <junit5.version>5.10.5</junit5.version>
         <maven.version>3.9.9</maven.version>

--- a/independent-projects/tools/pom.xml
+++ b/independent-projects/tools/pom.xml
@@ -57,7 +57,7 @@
         <mockito.version>5.15.2</mockito.version>
         <quarkus.version>${project.version}</quarkus.version>
         <maven-model-helper.version>37</maven-model-helper.version>
-        <jandex.version>3.2.4</jandex.version>
+        <jandex.version>3.2.5</jandex.version>
         <system-stubs-jupiter.version>2.0.2</system-stubs-jupiter.version>
         <awaitility.version>4.2.2</awaitility.version>
         <java-properties.version>0.0.7</java-properties.version>

